src/system/clients/client/secure_storage_client/secure_storage_client.rs

```diff
@@ -0,0 +1,482 @@
+    }
+
+    /// Save `profile` and saves its id as active profile id
+    pub async fn save_profile_and_active_profile_id(
+        &self,
+        profile: &Profile,
+    ) -> Result<()> {
+        debug!(
+            "Saving profile, id: {}, and setting it as active",
+            &profile.id()
+        );
+        self.save_profile(profile).await?;
+        self.save_active_profile_id(profile.id()).await
+    }
+
+    /// Save `profile`
+    pub async fn save_profile(&self, profile: &Profile) -> Result<()> {
+        let profile_id = profile.id();
+        debug!("Saving profile with id: {}", profile_id);
+        self.save(SecureStorageKey::ProfileSnapshot { profile_id }, profile)
+            .await
+            .inspect(|_| debug!("Saved profile with id {}", profile_id))
+            .inspect_err(|e| error!("Failed to save profile, error {e}"))
+    }
+
+    /// Save `profile_id` as the active profile id
+    pub async fn save_active_profile_id(
+        &self,
+        profile_id: ProfileID,
+    ) -> Result<()> {
+        debug!("Saving active profile id: {}", profile_id);
+        self.save(SecureStorageKey::ActiveProfileID, &profile_id)
+            .await
+            .inspect(|_| debug!("Saved active profile id"))
+            .inspect_err(|e| {
+                error!("Failed to save active profile id, error {e}")
+            })
+    }
+
+    //======
+    // DeviceInfo CR(U)D
+    //======
+
+    /// Loads the  DeviceInfo if any
+    pub async fn load_device_info(&self) -> Result<Option<DeviceInfo>> {
+        trace!("Loading device info");
+        self.load(SecureStorageKey::DeviceInfo).await
+    }
+
+    /// Saves [`DeviceInfo`]
+    pub async fn save_device_info(
+        &self,
+        device_info: &DeviceInfo,
+    ) -> Result<()> {
+        debug!("Saving new device info: {:?}", device_info);
+        self.save(SecureStorageKey::DeviceInfo, device_info)
+            .await
+            .inspect(|_| debug!("Saved new device info."))
+            .map_err(|e| {
+                error!(
+                    "Failed to save device info to secure storage - error {e}",
+                );
+                CommonError::UnableToSaveDeviceInfoToSecureStorage
+            })
+    }
+
+    //======
+    // Mnemonic CR(U)D
+    //======
+
+    /// Saves the MnemonicWithPassphrase of the private hd factor source
+    pub async fn save_private_hd_factor_source(
+        &self,
+        private_hd_factor_source: &PrivateHierarchicalDeterministicFactorSource,
+    ) -> Result<()> {
+        self.save_mnemonic_with_passphrase(
+            &private_hd_factor_source.mnemonic_with_passphrase,
+            &private_hd_factor_source.factor_source.id,
+        )
+        .await
+    }
+
+    /// Saves a MnemonicWithPassphrase under a given `FactorSourceIDFromHash`
+    pub async fn save_mnemonic_with_passphrase(
+        &self,
+        mnemonic_with_passphrase: &MnemonicWithPassphrase,
+        id: &FactorSourceIDFromHash,
```

Comment:
sure could have! but taking by ref makes the API a bit more flexible I think, because of `partial move`. Imaging this:

```rust
struct IdentifiableObject {
   id: ID,
   field2: SomeType,
}

fn with_id(id: &ID) { ... }
fn take_object(obj: IdentifiableObject)

fn frobnicate(obj: IdentifiableObject) {
   with_id(&obj.id) // we have not moved `obj` since we used `id` by ref (borrowed)
   take_object(obj) // free to move `obj` here.
}
```

Versus:

```rust
struct IdentifiableObject  {}  // same as above
fn with_id(id: ID) { ... } // NOT by ref here
fn take_object // same as above

fn frobnicate(obj: IdentifiableObject) {
   with_id(obj.id.clone()) // we MUST clone to not consume id
   take_object(obj) // now we are free to move `obj` here, since we cloned id above.
}
```

Imagine then ID is somewhat large, then it is more expensive to clone than take by ref (borrow).

Reply:
Thanks for explanation! So in this case it is basically for potential other usages of `FactorSourceIDFromHash` after calling `save_mnemonic_with_passphrase`